native/packages/react-native-bpk-component-image/stories.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Component } from 'react';
+import { View, ScrollView, Image, StyleSheet } from 'react-native';
+import BpkButton from 'react-native-bpk-component-button';
+import { spacingBase, spacingLg } from 'bpk-tokens/tokens/base.react.native';
+import { storiesOf } from '@storybook/react-native';
+
+import CenterDecorator from '../../storybook/CenterDecorator';
+import BpkImage from './index';
+
+const styles = StyleSheet.create({
+  button: {
+    marginBottom: spacingBase,
+  },
+  image: {
+    width: '100%',
+    height: spacingLg * 10,
+  },
+});
+
+class BpkImageLoadingStory extends Component<Props, State> {
+  static propTypes = {};
+
+  static defaultProps = {};
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      loaded: false,
+    };
+  }
+
+  onLoadToggle = () => {
+    this.setState(prevState => ({ loaded: !prevState.loaded }));
+  };
+
+  render() {
+    return (
+      <View>
+        <BpkButton
+          style={styles.button}
+          title={
+            this.state.loaded ? 'Reload story to try again' : 'Mark as loaded'
+          }
+          onPress={this.onLoadToggle}
+          disabled={this.state.loaded}
+        />
+        <BpkImage
+          loaded={this.state.loaded}
+          alt="test"
+          style={styles.image}
+          {...this.props}
+        />
+      </View>
+    );
+  }
+}
+
+storiesOf('react-native-bpk-component-image', module)
+  .addDecorator(CenterDecorator)
+  .add('docs:default', () => (
+    <BpkImage
+      alt="test"
+      source={{
+        uri:
+          'https://content.skyscnr.com/96508dbac15a2895b0147dc7e7f9ad30/canadian-rockies-canada.jpg',
+      }}
+      style={styles.image}
+    />
+  ))
+  .add('docs:no-border-radius', () => (
+    <BpkImage
+      rounded={false}
+      alt="test"
+      source={{
+        uri:
+          'https://content.skyscnr.com/200946ddb82b7c026e6e186a7037b1f8/machu-picchu.jpg',
+      }}
+      style={styles.image}
+    />
+  ))
+  .add('docs:loading', () => (
+    <BpkImageLoadingStory
+      source={{
+        uri:
+          'https://content.skyscnr.com/8a8ac17b591b61e6fe5d8f63414561cd/amsterdam-the-netherlands.jpg',
+      }}
+    />
+  ))
+  .add('performance-test-Image', () => (
```

Comment:
nit: use perf:{desc} the same way we use docs:{desc}. 

I also don't see the point of testing the base Image component here. Is it to compare it with our component? If so, I think I can be removed, I imagine the base component will be faster and that's ok, we only need to check if ours is fast enough.

Reply:
That's exactly why it's there. I don't think it's a bad thing for it to stay. At the moment `BpkImage` (once fully rendered) uses no more views than `Image`. If someone changes the component in the future they should check the same again. As there is still a lot of work to do on this component (lazy loading etc) I think it's worth leaving in

Reply:
I'd agree if this was actually measuring and comparing both, and if keeping the same performance as the base image component was a prerequisite. But it's bit hard to manually test both and decide if they're performing the same way. 

We really only need to check if our implementation is keeping 60 fps and not dropping frames. Also, we are actually testing the performance of the `ScrollView` here, it would be different we use a `RecyclerView`, for example.

The implementation seems simple enough to require a performance test at the moment, this seems a bit of premature optimisation IMO.

Reply:
Yeah actually I agree!